planning/sampling_based_planner/path_sampler/src/node.cpp

```diff
@@ -377,65 +390,89 @@ std::vector<TrajectoryPoint> PathSampler::generateTrajectory(const PlannerData &
-    const auto closest_iter = std::min_element(
-      prev_path_->points.begin(), prev_path_->points.end() - 1,
-      [&](const auto & p1, const auto & p2) {
-        return boost::geometry::distance(p1, current_state.pose) <=
-               boost::geometry::distance(p2, current_state.pose);
-      });
-    if (
-      closest_iter != prev_path_->points.end() &&
-      boost::geometry::distance(*closest_iter, current_state.pose) <= max_deviation) {
-      const auto current_idx = std::distance(prev_path_->points.begin(), closest_iter);
-      const auto length_offset = prev_path_->lengths[current_idx];
-      for (auto & l : prev_path_->lengths) l -= length_offset;
-      constexpr auto behind_dist = -5.0;  // [m] TODO(Maxime): param
-      auto behind_idx = current_idx;
-      while (behind_idx > 0 && prev_path_->lengths[behind_idx] > behind_dist) --behind_idx;
-      *prev_path_ = *prev_path_->subset(behind_idx, prev_path_->points.size());
-
-      // Use previous path for replanning
-      const auto prev_path_length = prev_path_->lengths.back();
-      const auto reuse_step = prev_path_length / params_.sampling.previous_path_reuse_points_nb;
-      for (double reuse_length = reuse_step; reuse_length <= prev_path_length;
-           reuse_length += reuse_step) {
-        sampler_common::State reuse_state;
-        size_t reuse_idx = 0;
-        for (reuse_idx = 0; reuse_idx + 1 < prev_path_->lengths.size() &&
-                            prev_path_->lengths[reuse_idx] < reuse_length;
-             ++reuse_idx)
-          ;
-        if (reuse_idx == 0UL) continue;
-        const auto reused_path = *prev_path_->subset(0UL, reuse_idx);
-        reuse_state.curvature = reused_path.curvatures.back();
-        reuse_state.pose = reused_path.points.back();
-        reuse_state.heading = reused_path.yaws.back();
-        reuse_state.frenet = path_spline.frenet(reuse_state.pose);
-        auto paths = generateCandidatePaths(reuse_state, path_spline, reuse_length, params_);
-        for (auto & p : paths) candidate_paths.push_back(reused_path.extend(p));
-      }
+sampler_common::Path PathSampler::generatePath(const PlannerData & planner_data)
+{
+  time_keeper_ptr_->tic(__func__);
+  sampler_common::Path generated_path{};
+
+  if (prev_path_ && prev_path_->points.size() > 1) {
+    sampler_common::constraints::checkHardConstraints(*prev_path_, params_.constraints);
+    if (prev_path_->constraint_results.isValid()) {
+      const auto prev_path_spline =
+        preparePathSpline(trajectory_utils::convertToTrajectoryPoints(*prev_path_), false);
+      const auto frenet_p = prev_path_spline.frenet(
+        {planner_data.ego_pose.position.x, planner_data.ego_pose.position.y});
+      if (std::abs(frenet_p.d) > params_.path_reuse.max_lat_dev || frenet_p.s < 0.0)
+        resetPreviousData();
+      else if (frenet_p.s <= params_.path_reuse.direct_reuse_dist)
+        return *prev_path_;
```

Comment:
I think that  if (frenet_p.s > params_.path_reuse.direct_reuse_dist) is true then resetPreviousData() is done regardless if 
(std::abs(frenet_p.d) > params_.path_reuse.max_lat_dev || frenet_p.s < 0.0) is true or not right? so we can eliminate the last condition?
```suggestion
    if (frenet_p.s > params_.path_reuse.direct_reuse_dist){
    resetPreviousData();
    }
     else{   return *prev_path_;}
        
        resetPreviousData();
        
```

Reply:
Thank you for your review.
This cannot be transformed into an `if...else...` clause because there are cases where we do not want to execute any of the two branches. For example if `frenet_p.s = 10.0`, `frenet_p.d = 0.0`, `params_.path_reuse.max_lat_dev = 2.0`, and `params_.path_reuse.direct_reuse_dist = 5.0`.